trl/cli.py

```diff
@@ -45,8 +45,13 @@ def main():
     make_sft_parser(subparsers)
     make_vllm_serve_parser(subparsers)
 
-    # Parse the arguments
-    args = parser.parse_args()
+    # Check if --config is in the arguments
+    if "--config" in sys.argv:
+        # Use the custom parse_args_and_config method to handle the config file
+        args = parser.parse_args_and_config()[0]
+    else:
+        # Parse the arguments normally if no config file is specified
+        args = parser.parse_args()
```

Comment:
maybe we should check here that `dataset_name` is in args? I needed to make `dataset_name` optional so that a call to `parser.parse_args()` works in the cases further down, checking that args contains `dataset_name` will then make sure that the same checks are in place as without this PR.